app/kumactl/cmd/install/context/install_control_plane_context.go

```diff
@@ -97,8 +99,19 @@ func DefaultInstallCpContext() InstallCpContext {
 			Ingress_service_type:                    "LoadBalancer",
 		},
 		InstallCpTemplateFiles: func(args *InstallControlPlaneArgs) (data.FileList, error) {
-			return data.ReadFiles(deployments.KumaChartFS())
+			files, err := data.ReadFiles(deployments.KumaChartFS())
+			if err != nil {
+				return nil, err
+			}
+			if !args.ExperimentalGateway {
+				files = files.Filter(ExcludeGatewayCRDs)
+			}
+			return files, nil
 		},
 		HELMValuesPrefix: "",
 	}
 }
+
+func ExcludeGatewayCRDs(file data.File) bool {
+	return !strings.HasPrefix(file.Name, "kuma.io_gateway")
```

Comment:
can we explicitly filter the CRDs we don't want? The name doesn't necessarily have anything to do with whether it's related to the Gateway feature

Reply:
makes sense